elpy/blackutil.py

```diff
@@ -43,7 +51,10 @@ def fix_code(code, directory):
     # Get black config from pyproject.toml
     line_length = black.DEFAULT_LINE_LENGTH
     string_normalization = True
-    pyproject_path = os.path.join(directory, "pyproject.toml")
+    if find_pyproject_toml:
+        pyproject_path = find_pyproject_toml((directory,))
```

Comment:
I _**think**_ this may be the root cause of #1955. It _**looks**_ like (at least with `black==21.12b0` that `find_pyproject_toml` will return `None` if it can't find `"pyproject.toml"` in `directory`.

Calling `os.path.exists(pyproject_path)` then equates to `os.path.exists(None)`, which will raise an exception:

``` python
>>> import os.path
>>> os.path.exists(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/MacPorts-20211130/Python.framework/Versions/3.9/lib/python3.9/genericpath.py", line 19, in exists
    os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
```